examples/campaign_management/add_complete_campaigns_using_batch_job.rb

```diff
@@ -0,0 +1,370 @@
+def build_campaign_budget_operation(client, customer_id)
+  # Creates a campaign budget operation.
+  operation = client.operation.create_resource.campaign_budget do |b|
+    # Creates a resource name using the temporary ID.
+    b.resource_name = client.path.campaign_budget(
+      customer_id, get_next_temporary_id)
+    b.name = "Interplanetary Cruise Budget ##{(Time.new.to_f * 1000).to_i}"
+    b.delivery_method = :STANDARD
+    b.amount_micros = 5_000_000
+  end
+
+  operation
+end
+
+# Builds new campaign operations for the specified customer ID.
+def build_campaign_operations(client, customer_id, campaign_budget_resource_name)
+  operations = []
+  for i in 0..NUMBER_OF_CAMPAIGNS_TO_ADD-1
+    # Creates a campaign.
+    campaign_id = get_next_temporary_id
+    operations << client.operation.create_resource.campaign do |c|
+      c.resource_name = client.path.campaign(customer_id, campaign_id)
+      c.name = "Mutate job campaign ##{(Time.new.to_f * 1000).to_i}.#{campaign_id}"
+      c.advertising_channel_type = :SEARCH
+      # Recommendation: Set the campaign to PAUSED when creating it to prevent
+      # the ads from immediately serving. Set to ENABLED once you've added
+      # targeting and the ads are ready to serve.
+      c.status = :PAUSED
+      # Sets the bidding strategy and budget.
+      c.manual_cpc = client.resource.manual_cpc
+      c.campaign_budget = campaign_budget_resource_name
+    end
+  end
+
+  operations
+end
+
+# Builds new campaign criterion operations for creating negative campaign
+# criteria (as keywords).
+def build_campaign_criterion_operations(client, campaign_operations)
+  operations = []
+  campaign_operations.each do |op|
+    operations << client.operation.create_resource.campaign_criterion do |cc|
+      cc.keyword = client.resource.keyword_info do |k|
+        k.text = "venus"
+        k.match_type = :BROAD
+      end
+      cc.negative = true
+      cc.campaign = op.create.resource_name
+    end
+  end
+
+  operations
+end
+
+# Builds new ad group operations for the specified customer ID.
+def build_ad_group_operations(client, customer_id, campaign_operations)
+  operations = []
+  campaign_operations.each do |op|
+    for i in 0..NUMBER_OF_AD_GROUPS_TO_ADD-1
+      # Creates an ad group.
+      ad_group_id = get_next_temporary_id
+      operations << client.operation.create_resource.ad_group do |ag|
+        # Creates a resource name using the temporary ID.
+        ag.resource_name = client.path.ad_group(customer_id, ad_group_id)
+        ag.name = "Mutate job ad group ##{(Time.new.to_f * 1000).to_i}.#{ad_group_id}"
+        ag.campaign = op.create.resource_name
+        ag.type = :SEARCH_STANDARD
+        ag.cpc_bid_micros = 10_000_000
+      end
+    end
+  end
+
+  operations
+end
+
+# Builds new ad group criterion operations for creating keywords. 50% of
+# keywords are created with some invalid characters to demonstrate how
+# MutateJobService returns information about such errors.
```

Comment:
Please replace `MutateJobService` with [`BatchJobService`](https://developers.google.com/google-ads/api/reference/rpc/v5/BatchJobService)

Reply:
Good catch. Changed to BatchJobService.